idaes/core/base/property_base.py

```diff
@@ -504,6 +505,68 @@ def report(self, index=(0), true_state=False, dof=False, ostream=None, prefix=""
 
         ostream.write("\n" + "=" * max_str_length + "\n")
 
+    def get_port_reference_name(self, component_name, port_name):
+        """
+        Get the standard name of a "port reference", the component
+        accessed on the port by "port_name.component_name".
+
+        Args:
+            component_name - name of port member of interest (str)
+            port_name - name of Port object (str)
+
+        Returns:
+            str with name for Reference used for Port member
+        """
+        return f"_{component_name}_{port_name}_ref"
+
+    def build_port(
+        self,
+        port_name,
+        doc=None,
+        slice_index=None,
+        index=None,
+    ):
+        """
+        Constructs a Port based on this StateBlock attached to the target block.
+
+        Args:
+            port_name - name to use for Port
+            doc - doc string or Prot object
+            slice_index - Slice index (e.g. (slice(None), 0.0) that will be
+                used to index self when constructing port references. DEfault = None.
+            index - time index to use when calling define_port_memebers. Default = None.
```

Comment:
Do we have an 80-character line length limit standard in IDAES? I would prefer to stay within this limit if possible, so that we don't have wrap-around in a standard terminal window with reasonable zoom.

Reply:
I think our line limit is a bit longer these days; we are now running `black` on the code so it will make sure everything has a standard line length.